File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03032/p03032_num4_parsed.py
import collections

def maximize_jewels(N, K, V):
    max_sum = 0
    
    for i in range(min(N, K) + 1):
        for j in range(min(N, K) + 1 - i):
            jewels = collections.deque(V[:i] + V[N - j:])
            
            while len(jewels) > 0 and jewels[0] < 0 and K - i - j > 0:
                jewels.popleft()
                K -= 1
            
            max_sum = max(max_sum, sum(jewels))
    
    return max_sum

N, K = map(int, input().split())
V = list(map(int, input().split()))

print(maximize_jewels(N, K, V))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03032/p03032_num4_parsed.cpp
#include <iostream>
#include <deque>
#include <algorithm>
using namespace std;

int main() {
    int N, K;
    cin >> N >> K;

    deque<int> D;
    for (int i = 0; i < N; i++) {
        int jewel;
        cin >> jewel;
        D.push_back(jewel);
    }

    int max_sum = 0;
    for (int A = 0; A <= min(N, K); A++) {
        for (int B = 0; B <= min(N - A, K - A); B++) {
            deque<int> hand;
            for (int i = 0; i < A; i++) {
                hand.push_back(D[i]);
            }
            for (int i = 0; i < B; i++) {
                hand.push_back(D[N - 1 - i]);
            }
            sort(hand.begin(), hand.end());
            int sum = 0;
            for (int i = 0; i < hand.size(); i++) {
                if (i < K - A - B && hand[i] < 0) {
                    continue;
                }
                sum += hand[i];
            }
            max_sum = max(max_sum, sum);
        }
    }

    cout << max_sum << endl;

    return 0;
}